src/site/content/en/web-dev-basics-one/fetch-api-error-handling/index.md

```diff
@@ -0,0 +1,234 @@
+-  Use any other metadata, such as [Response.headers](https://developer.mozilla.org/docs/Web/API/Response/headers), to assess whether the response was successful.
+
+```js
+let response;
+
+try {
+  response = await fetch('https://httpbin.org/status/429');  
+} catch (error) {
+  console.log('There was an error', error);
+}
+
+// Uses the 'optional chaining' operator
+if (response?.ok) {
+  console.log('Use the response here!');
+} else {
+  console.log('HTTP Response Code: ' + response?.status)
+}
+```
+
+The best practice is to work with people in your organization and team to understand potential HTTP response status codes. Backend developers, developer operations, and service engineers can sometimes provide unique insight into possible edge cases that you might not anticipate.
+
+{% Aside %}
+Note from the author: I was recently scrolling through an infinite scrolling list on a website, trying to quickly reach the bottom. Eventually, the page seemed to break with no helpful messaging. Upon inspecting the network request via DevTools, it became clear that the HTTP status code error `429 Too Many Requests` was returned. In an ideal scenario, such internal errors are surfaced to the web page interface.
+{% endAside %}
+
+### When there is an error parsing the network response
+
+This code example demonstrates another type of error that can arise with parsing a response body. The [Response](https://developer.mozilla.org/docs/Web/API/Response) interface offers convenient methods to parse different types of data, such as text or JSON. In the following code, a network request is made to an HTTP testing service that returns an HTML string as the response body. However, an attempt is made to parse the response body as [JSON](https://developer.mozilla.org/docs/Web/API/Response/json), throwing an error.
+
+```js
+let json;
+
+try {
+  const response = await fetch('https://httpbin.org/html');
+  json = await response.json();
+} catch (error) {
+  if (error instanceof SyntaxError) {
+    // Unexpected token < in JSON 
+    console.log('There was a SyntaxError', error);
+  } else {
+    console.log('There was an error', error);
+  }
+}
+
+if (json) {
+  console.log('Use the JSON here!', json);
+}
+```
+
+You must prepare your code to take in a variety of response formats, and verify that an unexpected response doesn't break the web page for the user.
+
+Consider the following scenario: You have a remote resource that returns a valid JSON response, and it is parsed successfully with the `Response.json()` method. It may happen that the service goes down. Once down, a [500 Internal Server Error](https://developer.mozilla.org/docs/Web/HTTP/Status/500) is returned. If appropriate error-handling techniques are not used during the parsing of JSON, this could break the page for the user because an unhandled error is thrown.
```

Comment:
```suggestion
Consider the following scenario: You have a remote resource that returns a valid JSON response, and it is parsed successfully with the `Response.json()` method. It may happen that the service goes down. Once down, a [`500 Internal Server Error`](https://developer.mozilla.org/docs/Web/HTTP/Status/500) is returned. If appropriate error-handling techniques are not used during the parsing of JSON, this could break the page for the user because an unhandled error is thrown.
```